google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/CollectionReference.java

```diff
@@ -171,13 +170,13 @@ public void remove() {
    * Adds a new document to this collection with the specified data, assigning it a document ID
    * automatically.
    *
-   * @param fields A Map containing the data for the new document.
+   * @param fields A Map or a POJO containing the data for the new document.
    * @return An ApiFuture that will be resolved with the DocumentReference of the newly created
    *     document.
    * @see #document()
    */
   @Nonnull
-  public ApiFuture<DocumentReference> add(@Nonnull final Map<String, Object> fields) {
+  public ApiFuture<DocumentReference> add(@Nonnull final Object fields) {
```

Comment:
This looks like a breaking API change. Is that true? If so, this would require a major version bump.

Reply:
I don't see why this would be breaking - do you have an example?